q2_sample_classifier/classify.py

```diff
@@ -371,6 +371,12 @@ def confusion_matrix(output_dir: str,
     if vmax == 'auto':
         vmax = None
 
+    predictions = predictions.astype(str)
+
+    temp_truth = truth.to_series()
+    temp_truth = temp_truth.astype(str)
+    truth = qiime2.CategoricalMetadataColumn(temp_truth)
```

Comment:
The three lines ending here shouldn't be necessary, since `truth` is already a `CategoricalMetadataColumn`, which means it _must_ have `str` (or `object`?) dtype.

Did you run into a problem that required these lines? If so, please share.